src/Database/Esqueleto/PostgreSQL/Pgcrypto.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | This module contains functions specific to the [pgcrypto](https://www.postgresql.org/docs/current/pgcrypto.html) module
+--
+-- @since: 3.5.1
+
+module Database.Esqueleto.PostgreSQL.Pgcrypto
+    (HashAlgorithm(..),
+     toCrypt,
+     fromCrypt,
+    ) where
+
+import qualified Data.Text.Internal.Builder as TLB
+import Database.Esqueleto.Experimental (toPersistValue)
+import Database.Esqueleto.Internal.Internal
+
+{- pgcrypto hashing algorithms
+see: https://www.postgresql.org/docs/current/pgcrypto.html
+
+`bf` and `xdes` algorithms have an optional iterations count parameter. All limitations and considerations
+mentioned in the `pgcrypto` module documentation regarding iteration count apply. It is possible to supply
+an invalid iteration count, which will lead to an sql error.
+
+/Requires/ the pgcrypto module.
+
+@since 3.5.1
+-}
+data HashAlgorithm
+    = BF (Maybe Word)
+    | MD5
```

Comment:
i feel like we should be attaching a `{-# WARNING MD5 "MD5 is considered insecure. Consider switching to another password hashing algorithm." #-}` so that folks don't try to use it. cf [What Is MD5 and Why Is It Considered Insecure?](https://www.section.io/engineering-education/what-is-md5/#:~:text=Unfortunately%2C%20MD5%20has%20been%20cryptographically,not%20be%20used%20for%20anything.&text=It%20is%20always%20recommended%20to,in%20any%20modern%20programming%20framework.)